sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,369 @@
+        t1_sel = r_sel.x[2]
+        t1_nonsel = r_nonsel.x[2]
+        
+        rCBF = 100 * (t1_nonsel - t1_sel) / t1_nonsel
+        CBF = multiplier * lambda_blood * (
+              (t1_nonsel / t1_blood) * ((1 / t1_sel) - (1 / t1_nonsel)))
+     
+        if outtype == 'simple':
+            r = [x for x in r_sel.x] + [x for x in r_nonsel.x] + [rCBF, CBF]
+            return [x if x is not None else 0 for x in r]
+        else: return [r_sel, r_nonsel, rCBF, CBF]
+    
+    else:
+        
+        if outtype == 'simple': return np.repeat(0, 8)
+        else:
+            return ['r_sel fit ' + r_sel.success,
+                    'r_nonsel fit ' + r_nonsel.success,
+                    'rCBF nan', 'CBF nan']
+
+
+def perf_fair_fitter_mp(all_s0, t1_blood, lambda_blood, ti, multiplier,
+                        t1_guess, picker_sel, picker_nonsel, ncpu):
+    """
+    Wrapper to execute perf_fair_fitter (outtype='simple') in parallel on
+    multiple signal vectors, tracking execution with a progress bar.
+    
+    Parameters
+    ----------
+    all_s0 : 2D numpy array of int or float
+        The aquired signals. The first dimension is the number of s0 vectors,
+        the second dimension the length of all of them. So for a real image,
+        usually len(D1) >> len(D2).
+        
+    t1_blood : int or float
+        T1 of blood in ms at the acquisition field strength. Empirical
+        constant; an example value is 2800 at 11.7T (Duong ref).
+        
+    lambda_blood : int or float
+        The assumed blood tissue partition coefficient of water in ml per g.
+        Empirical constant; usually 0.9.
+        
+    ti : list of int or float
+        The inversion times. Must have the same length as s0.
+        
+    multiplier : int or float
+        The absolute CBF result is initially produced in units of ml per g per
+        ms. The multiplier converts to desired units. Usually it is 6000000 to
+        convert to ml per 100g per min.    
+        
+    t1_guess : int or float
+        An initial starting value for T1. The mean of TIs is a good guess.
+    
+    picker_sel : list or numpy array of int
+        Vector indicating positions of selectively inverted signals in s0. The
+        lengths of picker_sel and picker_sel must sum to the length of s0.
+        
+    picker_nonsel : list or numpy array of int
+        Vector indicating positions of non-selectively inverted signals in s0.
+        The lengths of picker_sel and picker_sel must sum to the length of s0.
+        
+    ncpu : int
+        Number of processes to launch in parallel.    
+    
+    Returns
+    -------
+    List of all_s0.shape[0] lists, each of 8 floats (see perf_fair_fitter).
+    """
+    
+    # inspiration provided by:
+    # https://stackoverflow.com/q/5442910
+    # https://stackoverflow.com/a/45276885
+    
+    # if __name__ == '__main__': this might be dangerous
+    if 2 > 1: # to keep the indent in case above line is reinstated

```

Comment:
what do you mean ?

Reply:
It was quicker to write than changing the block indentation due to commenting out the previous line. Will delete it